ql/src/java/org/apache/hadoop/hive/ql/ddl/table/info/TableInfoUtils.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.ql.ddl.table.info;
+
+import java.util.Map;
+
+import org.apache.hadoop.hive.ql.ErrorMsg;
+import org.apache.hadoop.hive.ql.ddl.table.partition.PartitionUtils;
+import org.apache.hadoop.hive.ql.metadata.Hive;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.apache.hadoop.hive.ql.metadata.Table;
+import org.apache.hadoop.hive.ql.parse.SemanticException;
+
+/**
+ * Utilities used by table information DDL commands.
+ */
+public final class TableInfoUtils {
+  private TableInfoUtils() {
+    throw new UnsupportedOperationException("TableInfoUtils should not be instantiated");
+  }
+
+  public static void validateDatabase(Hive db, String databaseName) throws SemanticException {
+    try {
+      if (!db.databaseExists(databaseName)) {
+        throw new SemanticException(ErrorMsg.DATABASE_NOT_EXISTS.getMsg(databaseName));
+      }
+    } catch (HiveException e) {
+      throw new SemanticException(ErrorMsg.DATABASE_NOT_EXISTS.getMsg(databaseName), e);
+    }
+  }
+
+  public static void validateTable(Hive db, Table table, Map<String, String> partSpec) throws SemanticException {
+    if (partSpec != null) {
```

Comment:
what does this method validate? it only adds a nullcheck...I mean it doesn't even throw an exception ; I would expect that...from the word "validate"

Reply:
It validates that the partition exists, if there is a partition specified. It throws a SemanticException if the partition doesn't exist.

Reply:
It will be moved to Hive.

Reply:
Actually no need to move it to Hive, PartitionUtils can be called directly from wherever it is needed.

Reply:
Fixed.